### PR TITLE
Add JSON encoding to all applicable Web API arguments

### PR DESCRIPTION
**Pull request by jmanian**
There was at least one `chat.*` method that was missing the auto-encoding of `blocks` and `attachments` into JSON (`chat.scheduleMessage`). Instead of continuing to apply these to each method individually as a patch, I thought it would be nice to solve this once and for all.

~New groups of methods or new options that require encoding will require changes to accommodate those, but no patching.~ This is now being pulled from data in [slack-api-ref](https://github.com/slack-ruby/slack-api-ref/) and will update automatically with the docs.

~I'm open to suggestions on the exact approach, and I'm opening this as a draft without tests, changelog, etc. to gather comments first.~

### CODE REVIEW

File: lib/slack/web/api/templates/method.erb [hunk 1]
```diff
@@ -51,6 +51,9 @@ module Slack
 <% if data['deprecated'] %>
             logger.warn('<%= group %>.<%= name %>: <%= data['deprecation']['deprecation_warning']%> Alternative methods: <%= data['deprecation']['alternative_methods'].join(', ')%>.')
 <% end %>
+<% if %w[chat views dialog].include?(group) %>
```
Comment (jmanian) [inline lib/slack/web/api/templates/method.erb:54]: I don't love that this array of strings will need to be updated in coordination with `encode_options` when new groups are added, but I also wasn't keen to add this line to all endpoints.

File: lib/slack/web/api/options.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          when 'dialog'
+            encode_json(options, :dialog)
+          else
+            options
+          end
+        end
+
+        private
+
+        def encode_json(options, key)
```
Comment (dblock) [inline lib/slack/web/api/options.rb:21]: Make `encode_json` take an array, so

```ruby
def encode_json(options, keys)
   Array(keys).each ...
     if ...
        options = 
     end
   end
   options
end
```

File: lib/slack/web/api/templates/method.erb (hunk 1 above)
Comment (dblock) [reply lib/slack/web/api/templates/method.erb:54]: Can we promote `encoding: json` or a similar flag into slack-api-ref to avoid this?

**Review by dblock — COMMENTED** [review]
I like this very much! See my suggestions/comments below.

**Comment by jmanian** [comment]
@dblock Here's a different take, trying to base it entirely on data in `slack-api-ref`. It feels a little fragile to base it on data in the argument descriptions, but there isn't anything more structured at the moment (in `slack-api-ref` or on the doc pages). It does find all the ones we knew about, plus a handful of others.

**Comment by jmanian** [comment]
@dblock still needs tests and changelog, but here's what this looks like now with the slack-ruby/slack-api-ref#64

**Comment by dblock** [comment]
Looks good!

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ### 2.1.1 (Next)
```
Comment (dblock) [inline CHANGELOG.md:1]: Let's bump the version to 2.2 and make a release?

Comment (jmanian) [reply CHANGELOG.md:1]: Makes sense. I'm going to put up another PR to replace some of the patched argument validations with the new `arg_groups` data from the ref, so I'll do it in that.

**Pull request merged**